src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -56,18 +56,18 @@ TaskRevolutionParameters::TaskRevolutionParameters(PartDesignGui::ViewProvider*
     QMetaObject::connectSlotsByName(this);
     this->groupLayout()->addWidget(proxy);
 
-    //bind property mirrors
-    PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+    // bind property mirrors
+    auto pcFeat = dynamic_cast<PartDesign::ProfileBased *>(vp->getObject());

```

Comment:
Use `auto*` instead of `auto` to make it clear that a pointer is used.

Replacing static_cast with dynamic_cast here is OK because from the view provider type `PartDesignGui::ViewProvider` it's not obvious that the object type should be `PartDesign::ProfileBased`. But you should also check that `pcFeat` is not null as otherwise a segfault is still possible.

Reply:
done